utils_types.py

```diff
@@ -0,0 +1,11 @@
+from typing import Generator
```

Comment:
We try to avoid file names and directories that are called `utils`. It is often too general and doesn't really mean anything. 
Let's try giving this module a better name.